sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -1107,14 +1111,24 @@ impl ty::TyExpression {
         // Check if this could be a function
         let mut function_probe_warnings = Vec::new();
         let mut function_probe_errors = Vec::new();
-        let maybe_function = {
+
+        let maybe_function_decl = {
```

Comment:
I think this being a `maybe_` is confusing since those are used in the match to differentiate the types and it looks like it's only relevant in the function case.

The decl fetching should probably be moved into the `maybe_function` block and then later actually use `_fun_decl` instead of ignoring it.

Moreover, though that's out of scope, this entire function should probably be broken up so there's a clear interface and enforced pattern instead of mixing scopes.